include/deal.II/matrix_free/vector_access_internal.h

```diff
@@ -24,6 +24,8 @@
 #include <deal.II/matrix_free/dof_info.h>
 #include <deal.II/matrix_free/type_traits.h>
 
+#include <boost/algorithm/string/join.hpp>
```

Comment:
How much code does this include file pull in?

Reply:
It's also a borderline use case that can easily be done differently, in case there is a huge amount of code getting in.

Reply:
https://www.boost.org/doc/libs/1_50_0/boost/algorithm/string/join.hpp. I have added an ifdef so that it is only included in release mode. Would that work for you?